### PR TITLE
vendor: Bump cAdvisor to v0.45.0

### PR DESCRIPTION
**Pull request by bobbypage**
Signed-off-by: David Porter <david@porter.me>

#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

This updates to the latest version of cAdvisor.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

cAdvisor diff - https://github.com/google/cadvisor/compare/v0.44.1...v0.45.0

This version includes the containerd API vendored directly into cAdvisor as part of https://github.com/containerd/containerd/issues/7231 (https://github.com/google/cadvisor/pull/3145 by @dims) to ensure that k/k is not vendoring an EoL version of containerd. 

Note: containerd is still listed as dependency in `go.mod` because `hcsshim` requires it. 

#### Does this PR introduce a user-facing change?

```release-note
Update cAdvisor to v0.45.0
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -155,7 +155,6 @@ require (
 	github.com/cilium/ebpf v0.7.0 // indirect
 	github.com/containerd/cgroups v1.0.1 // indirect
 	github.com/containerd/console v1.0.3 // indirect
-	github.com/containerd/containerd v1.4.12 // indirect
```
Comment (liggitt) [inline go.mod:158]: 🎉

File: go.mod [hunk 2]
```diff
@@ -375,6 +374,7 @@ replace (
 	github.com/go-task/slim-sprig => github.com/go-task/slim-sprig v0.0.0-20210107165309-348f09dbbbc0
 	github.com/godbus/dbus/v5 => github.com/godbus/dbus/v5 v5.0.6
 	github.com/gofrs/uuid => github.com/gofrs/uuid v4.0.0+incompatible
+	github.com/gogo/googleapis => github.com/gogo/googleapis v1.4.1
```
Comment (liggitt) [inline go.mod:377]: is this new? where did this come from?

Comment (bobbypage) [reply go.mod:377]: See https://github.com/google/cadvisor/pull/3145#discussion_r934795153

It is needed as part of pulling in the new docker, I believe this was the PR where it was added - https://github.com/moby/moby https://github.com/moby/moby/pull/43239/

File: go.mod (hunk 1 above)
Comment (bobbypage) [reply go.mod:158]: Props to @dims for working on this!

**Comment by liggitt** [comment]
/lgtm
/approve

```diff
 {
-	"directDependencies": 229,
+	"directDependencies": 228,
-	"transitiveDependencies": 261,
+	"transitiveDependencies": 262,
-	"totalDependencies": 322,
+	"totalDependencies": 323,
 	"maxDepthOfDependencies": 30
 }
```

as a follow-up, might be worth adding containerd/containerd to our unwanted-dependencies.json file, and seeing if there's a way to avoid the gogo/googleapis transitive dep

File: go.mod (hunk 2 above)
Comment (dims) [reply go.mod:377]: indeed!

File: go.mod (hunk 1 above)
Comment (dims) [reply go.mod:158]: yay!

**Comment by dims** [comment]
@bobbypage looks like something changed underneath. please see the failure in pull-kubernetes-dependencies

NOTE for release team: This one had all the approvals to merge, but one of the things that merged before this one must have caused a dependency failure. @bobbypage will respin this and we will land it hopefully today.

**Comment by liggitt** [comment]
looks like this needs a hack/update-vendor.sh to fix the presubmit

**Comment by bobbypage** [comment]
Done, rebased and updated the missing dependency

**Comment by brandond** [comment]
Just wanted to cross-link https://github.com/google/cadvisor/issues/3204 from this PR.

**Comment by dims** [comment]
@brandond thanks for the feedback, linking to the slack thread you started too https://kubernetes.slack.com/archives/C0BP8PW9G/p1669844347437479

**Pull request merged**